### PR TITLE
Resolve version conflict react v15 vs v17

### LINKED ISSUE #4166: Try to launch locally fails with NPM
### Q&A (please complete the following information)
- OS: Ubuntu
- Browser: Chrome
- Version: 23
- Method of installation: docker
- Swagger-Editor version: latest

### Content & configuration
node 16.17.1

### Describe the bug you're encountering
The web application is not able to start

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: react-ace@4.4.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"=17.0.2" from the root project
npm ERR!   peer react@"^17.0.0-0" from @wojtekmaj/enzyme-adapter-react-17@0.8.0
npm ERR!   node_modules/@wojtekmaj/enzyme-adapter-react-17
npm ERR!     dev @wojtekmaj/enzyme-adapter-react-17@"=0.8.0" from the root project
npm ERR!   16 more (@wojtekmaj/enzyme-adapter-utils, boron, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^0.13.0 || ^0.14.0 || ^15.0.1" from react-ace@4.4.0
npm ERR! node_modules/react-ace
npm ERR!   react-ace@"^4.1.6" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@15.7.0
npm ERR! node_modules/react
npm ERR!   peer react@"^0.13.0 || ^0.14.0 || ^15.0.1" from react-ace@4.4.0
npm ERR!   node_modules/react-ace
npm ERR!     react-ace@"^4.1.6" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/9999/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:

### To reproduce...
nvm use 16.17.1
npm start (under swagger editor folder)

### Expected behavior
whatever else...

### PR DESCRIPTION
**Pull request by Janosch**
Update and remove outdated packages to resolve dependency conflict regarding react

### Description
The following packages require react < 17.0.2, but the project requires react=17.0.2. 

React-split-pane has been removed as it is outdated and it is apparently not used. The other packages have been updated to the most recent versions.

"react-ace": "^4.1.6",
"react-addons-css-transition-group": "^15.4.2"
"react-split-pane": "^0.1.82",
"react-transition-group": "^1.1.1",

### Motivation and Context
Resolve dependecy conflict
Fixes #4166

### How Has This Been Tested?
Opened Swagger editor locally

### My PR contains... 
- [x] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [x] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,14 +78,12 @@
     "prop-types": "15.8.1",
     "querystring-browser": "^1.0.4",
     "react": "=17.0.2",
-    "react-ace": "^4.1.6",
-    "react-addons-css-transition-group": "^15.4.2",
+    "react-ace": "^10.1.0",
```
Comment (char0n) [inline package.json:81]: SwaggerEditor will throw runtime errors and will not render with react-ace@10. What we can do is to update to `react-ace@7.0.5` as in this version there are no breaking changes. Any version above will require code changes and will break integrations and plugins of 3rd parties integrating with SwaggerEditor and expecting react-ace compatible API.

```suggestion
    "react-ace": "=7.0.5",
```

File: package.json [hunk 2]
```diff
@@ -32,7 +33,7 @@
     "deps-check:size": "webpack -p --config webpack/test_deps_size.babel.js --json | webpack-bundle-size-analyzer >| $npm_package_config_deps_check_dir/sizes.txt",
     "dev": "webpack serve --config webpack/dev.babel.js",
     "hot-server": "webpack serve --host 0.0.0.0 --config webpack/dev.babel.js",
-    "open-static": "node -e 'require(\"open\")(\"http://localhost:3001\")'",
+    "open-static": "open-cli http://localhost:3001",
```
Comment (char0n) [inline package.json:36]: Unnecessary change for this PR.

**Review by char0n — CHANGES_REQUESTED** [review]
Thanks for trying to solve some inconveniences while installing SwaggerEditor locally, but this PR has many issues and obviously haven't been tested locally at all.

**Comment by char0n** [comment]
I'm closing this one. If you want to help with this project please issue a PR with single dependency update so that we can approach updates progressively.

**Comment by Janosch** [comment]
> Thanks for trying to solve some inconveniences while installing SwaggerEditor locally, but this PR has many issues and obviously haven't been tested locally at all.

The changes I proposed were necessary for me to get Swagger Editor running in my environment. I am using the current LTS version of node (v18.14.2). I am using Swagger Editor with the changes in the PR to design an API as I am writing this, so it has been tested locally but not all scripts. `npm start` works for me.

I understand that for a PR there needs to be more thorough testing and all scripts need to work, so next time I will just point to the fork instead of creating a PR.

**Comment by char0n** [comment]
“npm start” is using build fragments from the “./dist” folder. This means no building is involved, only serving already prebuilt static files. That is why you have seen the editor working properly.
